### DISCUSSION TITLE
Issue with Serial Number on Xiaomi Dreame D9 Vacuum Robot during Valetudo Installation

### DISCUSSION THREAD

**Discussion by VaneukemY** [Hypfer/Valetudo#1819] [Q&A / Support] [answered]
Hello everyone,

While following the procedure to install Valetudo on my vacuum robot "Xiaomi Dreame D9", I noticed that I no longer had the label with the serial number of the robot behind the dustbin as indicated on the procedure (https://valetudo.cloud/pages/installation/dreame.html).

This serial number allows to know the password of the root user of the robot (as explained in the procedure). 

- Is there any other way to find the password? 
- Do you have a technique to find this serial number?

Thanks for your help in advance,
Youri.

**Comment by Hypfer** [comment, marked as answer]
You usually can deduce the SN in question from other SN stickers e.g. below the robot or below its dock.

>Dennis Giese, [27/12/2022 15:48]
>just for reference about the devices for the chinese market, that got relabeled for EU (or other regions). The dustbin SN does not match the SN in the Software. But you can get the "real" SN by looking at the bottom sticker (which seems to contain the real SN)
![image](https://github.com/Hypfer/Valetudo/assets/974410/8ea96c58-1653-4b23-961c-5c8a0e55e0fa)

**Reply by VaneukemY** [reply]
@Hypfer Thank you for your help!

**Comment by Cuel** [comment]
@VaneukemY this worked for you? My dustbin sticker is not visible anymore and I've tested with the full serial as well as the suggested image attached (and md5sum/b64) but it doesn't accept the password. Dreame D9.

**Reply by VaneukemY** [reply]
@Cuel I didn't test it yet, I can't root my vacuum robot due to its recent version.

**Comment by copardo** [comment]
I have the same issue. Lost the sticker and not able to login, tried aboves logic, but was not successfull. I have two stickers, one on the robots backside and one on the charger. Someone an idea, how my "password" could look like?

![WhatsApp Image 2025-01-06 at 21 19 18(1)](https://github.com/user-attachments/assets/ed033997-0a09-458f-9c35-f2d28e2e1003)
![WhatsApp Image 2025-01-06 at 21 19 18(2)](https://github.com/user-attachments/assets/93c92577-9b72-40f4-84e8-55deddbe266a)
![WhatsApp Image 2025-01-06 at 21 19 18](https://github.com/user-attachments/assets/6e857d55-1dcd-46c4-9015-7f13e78917ed)

**Comment by copardo** [comment]
Solved it! Next to the battery is another sticker which worked! You just need to take off the cover on the bottom

**Comment by opensource4life** [comment]
Hi, I have an unusual problem with rooting a **Dreame L10 Pro**. A few years ago, my friends and I each bought an L10, and I rooted them one by one. The last friend was afraid of losing the warranty (even though he bought it on Aliexpress from some random seller, so what warranty? ;)). Finally, he decided he wanted to disconnect the vacuum from the cloud and brought it to me. This model have a "worn-off" sticker with the serial number - the obvious one under the dust container and second located next to the battery. It looks like a deliberate action (but for what purpose?). I have the robot along with the docking station and the box; the robot's serial number from the sticker, the cardboard box, and the dock all match. I know that based on the serial number from the sticker, one can try to assemble the robot's serial number. So, having the sticker number **P20291408EU012450000P01**, after shortening it should be **P20291408EU01245ZM**, 
but the calculated password from this string does not allow logging in as root.

And here is the question: is there any method to extract the serial number? 

<img width="1200" height="1600" alt="bottom" src="https://github.com/user-attachments/assets/55043bcc-c3ad-4c77-9a7b-9113f3dd0b3e" />

**Reply by opensource4life** [reply]
![bottom2](https://github.com/user-attachments/assets/29300310-5ab4-4542-9e8b-7827a3fd9bda)
![chargestation](https://github.com/user-attachments/assets/a5f2a625-41fa-4428-ae61-8faa80eca28f)

**Reply by Hypfer** [reply]
Do not continue rooting that bot.

It might be one of those L/Z10 with a replacement PCB from RMA. Those are not rootable. Attempting to do so will brick them.

If you do not have the SN, do not continue.
